api/types/genargs.go

```diff
@@ -10,12 +10,12 @@ import (
 
 // GenArgs is a facade over GeneratorArgs, exposing a few readonly properties.
 type GenArgs struct {
-	args *GeneratorArgs
+	Args *GeneratorArgs
```

Comment:
According to the note above, the point of `GenArgs` is to expose only a select few properties. If we need to make `Args` public, does the separation still have a purpose?

Reply:
Probably not. Let me see if I can keep `args` private.

I ran into some implementation challenges that way (resources that were not generated were also getting hashes), but I think I will have the same challenges if we remove `GenArgs` as a wrapper.

Reply:
changed to keep args private